objects.go

```diff
@@ -168,6 +168,40 @@ func (o *Object) Clone() *Object {
 	return &objectCopy
 }
 
+// IsContainedInRect : returns true if the object
+// A primitive is in the rect if the entire primitive is
+// A Mesh without a primitive is in the rect if it's centre is
+func (o *Object) IsContainedInRect(rect Rect) bool {
```

Comment:
I think (the way they are written here) Object.IsContainedInRect and Primitive.IsContainedInRect do different things, and the names are confusing.

Object.IsContainedInRect checks if the origin of the object is within the rect.  So IsContainedInRect is a reasonable name, but IsOriginContainedInRect or IsOriginInRect are better names.  Since origin is a point, this could also be called OverlapsWithRect or OriginOverlapsWithRect.

Primitive.IsContainedInRect is just badly named IMO.  It should actually be OverlapsWithRect, since its not looking for containment, its looking for any overlap.